packages/core/src/sfpcommands/source/DeploySourceToOrgImpl.ts

```diff
@@ -215,72 +115,94 @@ export default class DeploySourceToOrgImpl implements DeploymentExecutor {
-            //testlevel
-            command += ` -l RunSpecifiedTests`;
-            apexclasses = this.convertApexTestSuiteToListOfApexClasses(this.deployment_options['apextestsuite']);
-            command += ` -r ${apexclasses}`;
+            metdataDeployOptions.apiOptions.testLevel = `RunSpecifiedTests`;
+            let apexTestSuite = new ApexTestSuite(sourceDir, this.deployment_options['apextestsuite']);
+            metdataDeployOptions.apiOptions.runTests = await apexTestSuite.getConstituentClasses();
         } else if (this.deployment_options['testlevel'] == 'RunSpecifiedTests') {
-            command += ` -l RunSpecifiedTests`;
-            apexclasses = this.deployment_options['specified_tests'];
-            command += ` -r ${apexclasses}`;
+            metdataDeployOptions.apiOptions.testLevel = `RunSpecifiedTests`;
+            metdataDeployOptions.apiOptions.runTests = this.deployment_options['specified_tests'].split(`,`);
         } else {
-            command += ` -l ${this.deployment_options['testlevel']}`;
+            metdataDeployOptions.apiOptions.testLevel = this.deployment_options['testlevel'];
         }
 
         if (this.deployment_options['ignore_warnings']) {
-            command += ` --ignorewarnings`;
+            metdataDeployOptions.apiOptions.ignoreWarnings = true;
         }
         if (this.deployment_options['ignore_errors']) {
-            command += ` --ignoreerrors`;
+            metdataDeployOptions.apiOptions.rollbackOnError = false;
         }
 
-        return command;
+        return metdataDeployOptions;
     }
 
-    private convertApexTestSuiteToListOfApexClasses(apextestsuite: string): Promise<string> {
-        SFPLogger.log(
-            `Converting an apex test suite  ${apextestsuite} to its consituent apex test classes`,
-            null,
-            this.packageLogger
-        );
+    convertApexTestSuiteToListOfApexClasses(apexTestSuite: any) {
+        throw new Error('Method not implemented.');
+    }
 
-        let result = child_process.execSync(
-            `sfdx sfpowerkit:source:apextestsuite:convert  -n ${apextestsuite} --json`,
-            { cwd: this.project_directory, encoding: 'utf8' }
-        );
+    private async deploy(backingSourceDir: string, componentSet: ComponentSet) {
+        let deploymentOptions = await this.buildDeploymentOptions(backingSourceDir, this.target_org);
+        const deploy = await componentSet.deploy(deploymentOptions);
+
+        let startTime = Date.now();
+        SFPLogger.log(`Deploying to ${this.target_org} with id:${deploy.id}`, LoggerLevel.INFO, this.packageLogger);
+        // Attach a listener to check the deploy status on each poll
+        deploy.onUpdate((response) => {
+            const { status, numberComponentsDeployed, numberComponentsTotal } = response;
+            const progress = `${numberComponentsDeployed}/${numberComponentsTotal}`;
+            const message = `Status: ${status} Progress: ${progress}`;
```

Comment:
progress should be a percentage

Reply:
dont think we need percentage, as its commonly denoted by number